nntrainer/include/properties.h

```diff
@@ -0,0 +1,363 @@
+// SPDX-License-Identifier: Apache-2.0-only
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file properties.h
+ * @date 03 Aug 2020
+ * @brief Properties handler for NNtrainer.
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#ifndef __NNTRAINER_PROPERTIES_H__
+#define __NNTRAINER_PROPERTIES_H__
+#ifdef __cplusplus
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <nntrainer_error.h>
+
+namespace nntrainer {
+template <typename T> struct prop_traits {
+  static constexpr bool is_property = false; /**< check if this is property */
```

Comment:
struct? class? What made you to use struct? Looks like it is ok to use class. Cause struct, class mixed and make codes more complicated